src/languages/singlestoredb/singlestoredb.formatter.ts

```diff
@@ -0,0 +1,267 @@
+  'DROP PROCEDURE',
+  'DROP RESOURCE POOL',
+  'DROP ROLE',
+  'DROP USER',
+  'DROP VIEW',
+  'EXECUTE',
+  'EXPLAIN',
+  'FLUSH',
+  'FORCE',
+  'GRANT',
+  'HANDLER',
+  'HELP',
+  'KILL CONNECTION',
+  'KILLALL QUERIES',
+  'LOAD DATA',
+  'LOAD INDEX INTO CACHE',
+  'LOAD XML',
+  'LOCK INSTANCE FOR BACKUP',
+  'LOCK TABLES',
+  'MASTER_POS_WAIT',
+  'OPTIMIZE TABLE',
+  'PREPARE',
+  'PURGE BINARY LOGS',
+  'REBALANCE PARTITIONS',
+  'RELEASE SAVEPOINT',
+  'REMOVE AGGREGATOR',
+  'REMOVE LEAF',
+  'REPAIR TABLE',
+  'REPLACE',
+  'REPLICATE DATABASE',
+  'RESET',
+  'RESET MASTER',
+  'RESET PERSIST',
+  'RESET REPLICA',
+  'RESET SLAVE',
+  'RESTART',
+  'RESTORE DATABASE',
+  'RESTORE REDUNDANCY',
+  'REVOKE',
+  'ROLLBACK',
+  'ROLLBACK TO SAVEPOINT',
+  'SAVEPOINT',
+  'SET CHARACTER SET',
+  'SET DEFAULT ROLE',
+  'SET NAMES',
+  'SET PASSWORD',
+  'SET RESOURCE GROUP',
+  'SET ROLE',
+  'SET TRANSACTION',
+  'SHOW',
+  'SHOW CHARACTER SET',
+  'SHOW COLLATION',
+  'SHOW COLUMNS',
+  'SHOW CREATE DATABASE',
+  'SHOW CREATE FUNCTION',
+  'SHOW CREATE PIPELINE',
+  'SHOW CREATE PROCEDURE',
+  'SHOW CREATE TABLE',
+  'SHOW CREATE USER',
+  'SHOW CREATE VIEW',
+  'SHOW DATABASES',
+  'SHOW ENGINE',
+  'SHOW ENGINES',
+  'SHOW ERRORS',
+  'SHOW FUNCTION CODE',
+  'SHOW FUNCTION STATUS',
+  'SHOW GRANTS',
+  'SHOW INDEX',
+  'SHOW MASTER STATUS',
+  'SHOW OPEN TABLES',
+  'SHOW PLUGINS',
+  'SHOW PRIVILEGES',
+  'SHOW PROCEDURE CODE',
+  'SHOW PROCEDURE STATUS',
+  'SHOW PROCESSLIST',
+  'SHOW PROFILE',
+  'SHOW PROFILES',
+  'SHOW RELAYLOG EVENTS',
+  'SHOW REPLICA STATUS',
+  'SHOW REPLICAS',
+  'SHOW SLAVE',
+  'SHOW SLAVE HOSTS',
+  'SHOW STATUS',
+  'SHOW TABLE STATUS',
+  'SHOW TABLES',
+  'SHOW VARIABLES',
+  'SHOW WARNINGS',
+  'SHUTDOWN',
+  'SNAPSHOT DATABASE',
+  'SOURCE_POS_WAIT',
+  'START GROUP_REPLICATION',
+  'START PIPELINE',
+  'START REPLICA',
+  'START SLAVE',
+  'START TRANSACTION',
+  'STOP GROUP_REPLICATION',
+  'STOP PIPELINE',
+  'STOP REPLICA',
+  'STOP REPLICATING',
+  'STOP SLAVE',
+  'TABLE',
+  'TEST PIPELINE',
+  'TRUNCATE TABLE',
+  'UNLOCK INSTANCE',
+  'UNLOCK TABLES',
+  'USE',
+  'WITH',
+  'XA',
+  // flow control
+  // 'IF',
+  'ITERATE',
+  'LEAVE',
+  'LOOP',
+  'REPEAT',
+  'RETURN',
+  'WHILE',
+  // other
+  'ADD',
+  'WINDOW',
+  'PARTITION BY',
+]);
+
+const reservedSetOperations = expandPhrases(['UNION [ALL | DISTINCT]']);
+
+const reservedJoins = expandPhrases([
+  'JOIN',
+  '{LEFT | RIGHT | FULL} [OUTER] JOIN',
+  '{INNER | CROSS} JOIN',
+  'NATURAL {LEFT | RIGHT} [OUTER] JOIN',
+  // non-standard joins
+  'STRAIGHT_JOIN',
+]);
+
+// https://dev.mysql.com/doc/refman/8.0/en/
+export default class SingleStoreDBFormatter extends Formatter {
+  static operators = ['=', '<=>', '>', '>=', '<', '<=', '!=', '<>'];
+
+  tokenizer() {
+    return new Tokenizer({
+      reservedCommands,
+      reservedSetOperations,
+      reservedJoins,
+      reservedDependentClauses: ['WHEN', 'ELSE', 'ELSEIF'],
```

Comment:
I found no example of `ELSEIF` syntax from SingeStoreDB docs.

Reply:
Will need to try it out...

Reply:
I have verified that `ELSEIF` is supported. Here is the example I have run:

```
DELIMITER // ;
CREATE OR REPLACE PROCEDURE courses_sp (course_code TEXT, section_number INT, number_students INT) AS   DECLARE
     code TEXT = UCASE(course_code); 
     num_students INT = number_students + 1;
     BEGIN
     IF '' = '' THEN INSERT INTO courses VALUES (code, section_number, num_students);
     ELSEIF 1 = 1 THEN INSERT INTO courses VALUES(code, section_number, num_students);
     END IF;
     END //
DELIMITER ; //
```
Here is the execution:
```
memsql> show procedures;
+-----------------------+------------------+---------+
| Procedures_in_example | Routine Lifetime | Definer |
+-----------------------+------------------+---------+
| courses_sp            | Permanent        | root@%  |
+-----------------------+------------------+---------+
1 row in set (0.001 sec)

memsql> CALL courses_sp("cs-101",1,13);
Query OK, 0 rows affected (0.504 sec)

memsql> select * from courses;
+-------------+----------------+-----------------+
| course_code | section_number | number_students |
+-------------+----------------+-----------------+
| CS-101      |              1 |              14 |
+-------------+----------------+-----------------+
1 row in set (0.192 sec)
```